sopel/privileges.py

```diff
@@ -1,146 +1,172 @@
-"""Constants for user privileges in channels.
+"""Constants for user privileges in channels."""
+# Copyright 2023 dgw, technobabbl.es
+from __future__ import annotations
 
-Privilege levels
-================
+import enum
 
-Historically, there were two user privileges in channels:
 
-* :data:`OP`: channel operator, or chanop, set and unset by ``+o`` and ``-o``
-* :data:`VOICE`: the privilege to send messages to a channel with the
-  ``+m`` mode, set and unset by ``+v`` and ``-v``
+__all__ = [
+    'AccessLevel',
+]
 
-Since then, other privileges have been adopted by IRC servers and clients:
 
-* :data:`HALFOP`: intermediate level between Voiced and OP, set and unset by
-  ``+h`` and ``-h``
-* :data:`ADMIN`: channel admin, above OP and below OWNER, set and unset by
-  ``+a`` and ``-a``
-* :data:`OWNER`: channel owner, above ADMIN and OP, set and unset by ``+q`` and
-  ``-q``
+class AccessLevel(enum.IntFlag):
+    """Enumeration of available user privilege levels.
 
-.. important::
+    Privilege levels
+    ================
```

Comment:
I think this should stay at the module level, as these explanations are not tied to the implementation details of "AccessLevel is an ENUM now". It was thought out as a standalone explanation for "what are privilege levels for Sopel", giving more context than the implementation details could.

Reply:
You make a lot of sense, as usual. I pulled this section out to the `docs/source/plugin/privileges.rst` intro section, before we get into how to work with privileges _within Sopel_; see b8c2d99e2890dc7f06ec8a65c0c8722779163472 and/or updated [deploy preview](https://deploy-preview-2540--sopel-unstable-docs.netlify.app/plugin/privileges#user-privileges). 🙂